src/main/java/org/synyx/urlaubsverwaltung/application/application/ApplicationForLeaveViewController.java

```diff
@@ -118,15 +119,24 @@ private void prepareApplicationModels(Model model, Locale locale) {
 
     private static ApplicationForLeaveDto toView(ApplicationForLeave application, Person signedInUser, MessageSource messageSource, Locale locale) {
         final Person person = application.getPerson();
+
         final boolean isWaiting = application.hasStatus(WAITING);
+        final boolean isAllowed = application.hasStatus(ALLOWED);
+        final boolean isTemporaryAllowed = application.hasStatus(TEMPORARY_ALLOWED);
         final boolean isCancellationRequested = application.hasStatus(ALLOWED_CANCELLATION_REQUESTED);
         final boolean twoStageApproval = application.isTwoStageApproval();
 
         final boolean isBoss = signedInUser.hasRole(BOSS);
+        final boolean isOffice = signedInUser.hasRole(OFFICE);
         final boolean isDepartmentHead = signedInUser.hasRole(DEPARTMENT_HEAD);
         final boolean isSecondStageAuthority = signedInUser.hasRole(SECOND_STAGE_AUTHORITY);
         final boolean isOwn = person.equals(signedInUser);
-        final boolean isAllowedToAllow = isBoss || ((isDepartmentHead || isSecondStageAuthority) && !isOwn);
+
+        final boolean isAllowedToEdit = isWaiting && isOwn;
+        final boolean isAllowedToTemporaryApprove = twoStageApproval && isWaiting && (isDepartmentHead && !isOwn) && !isBoss && !isSecondStageAuthority;
+        final boolean isAllowedToApprove = isWaiting && (isBoss || ((isDepartmentHead || isSecondStageAuthority) && !isOwn));
```

Comment:
```suggestion
        final boolean isAllowedToApprove = isWaiting && (isBoss || ((isSecondStageAuthority) && !isOwn));
```

DepartmentHead sollte nicht freigeben dürfen im zweistufigen Genehmigungsprozess